### DISCUSSION TITLE
High Resolution Local Map (Delete Nodes in Full Map To Bound Map Size)

### DISCUSSION THREAD

**Discussion by generush** [UnknownFreeOccupied/ufomap#5] [Ideas]
Hello,

I am interested in creating a high-resolution local map, something like 5-10-15 meter bounding box. The reason is that maintaining the entire map during a mobile robot mission (100m-1km) results in very slow publishing rates due to ballooning map size.

I have implemented some code to delete nodes outside the specified bounding box (relative to the robot position) in OctoMap ROS server, but I am only beginning to think about how to best implement that for ufomap.

Thanks for any and all ideas!

**Comment by generush** [comment]
Hi @danielduberg

I have found that the `update_part_of_map` parameter provides me exactly what I want, publisher for a locally constrained map. However, when I view the topic `/ufomap_mapping_server_node/map` in RViz, it seems to show the entire map, which is undesirable because it consumes a lot of memory, induces a lot of latency (making the visualization somewhat useless) and causes RViz to crash after 10-20 minutes.

I imagine what is happening, is that RViz is accumulating the messages across time, and visualizing all of them.

It would be great if there was a parameter to control whether this accumulation happens or not. For some folks, it might be desirable to keep all the map updates visualized. For my application, I would like to only see the most recent map update.

Thanks!

**Comment by huangyoupingben** [comment]
hi, @erush91
Did you fix this problem.
Maybe, I encountered a problem similar to yours. I want to use ufomap to build a local 3D map centered on a robot, like 20-20-8 meter bounding box. Howewer, When the robot  moves, the ufomap is stationary, not a  Robot-centric 3D map. I have tried **update_part_of_map** parameter, it did not work. 
Look forward to your reply.
Thanks!
![UFOMAP_is_globalMap (1)](https://user-images.githubusercontent.com/61875969/139017185-2e8a6e54-22d7-41de-826a-db6d6652da53.png)

**Comment by huangyoupingben** [comment]
hi,@erush91
I tried update_part_of_map parameter again,and changed some code, it worked.
But，still have a problem. Ufomap still needs to build a big map internally, and when the robot approaches the boundary of this big map, the local map may fail. 
![rviz_screenshot_2021_10_27-17_26_25](https://user-images.githubusercontent.com/61875969/139040283-7fa3dccc-6180-4e1e-bf3a-b0698c717fde.png)
![rviz_screenshot_2021_10_27-17_25_55](https://user-images.githubusercontent.com/61875969/139040287-25265df8-a5b6-4145-a27a-0673852c9e26.png)

**Reply by generush** [reply]
I am not familiar with your issue. Is that image the map or the point cloud? If it's the map, I would try to increase your map resolution to something like 5cm - 10cm and see if your problem persists.
